lib/src/date-input/Calendar.tsx

```diff
@@ -0,0 +1,495 @@
+      });
+    }
+    setDateToFocus(date);
+  }
+
+  const handleDayKeyboardEvent = (event, day) => {
+    let dateToFocusTemp = innerDate.set("date", day);
+    switch (event.key) {
+      case "PageUp":
+        event.preventDefault();
+        event.shiftKey
+          ? (dateToFocusTemp = dateToFocusTemp.set("year", dateToFocusTemp.get("year") - 1))
+          : (dateToFocusTemp = dateToFocusTemp.set("month", dateToFocusTemp.get("month") - 1));
+        focusDate(dateToFocusTemp);
+        break;
+      case "PageDown":
+        event.preventDefault();
+        event.shiftKey
+          ? (dateToFocusTemp = dateToFocusTemp.set("year", dateToFocusTemp.get("year") + 1))
+          : (dateToFocusTemp = dateToFocusTemp.set("month", dateToFocusTemp.get("month") + 1));
+        focusDate(dateToFocusTemp);
+        break;
+      case "ArrowLeft":
+        event.preventDefault();
+        dateToFocusTemp = dateToFocusTemp.set("date", dateToFocusTemp.get("date") - 1);
+        focusDate(dateToFocusTemp);
+        break;
+      case "ArrowRight":
+        event.preventDefault();
+        dateToFocusTemp = dateToFocusTemp.set("date", dateToFocusTemp.get("date") + 1);
+        focusDate(dateToFocusTemp);
+        break;
+      case "ArrowUp":
+        event.preventDefault();
+        dateToFocusTemp = dateToFocusTemp.set("date", dateToFocusTemp.get("date") - 7);
+        focusDate(dateToFocusTemp);
+        break;
+      case "ArrowDown":
+        event.preventDefault();
+        dateToFocusTemp = dateToFocusTemp.set("date", dateToFocusTemp.get("date") + 7);
+        focusDate(dateToFocusTemp);
+        break;
+      case "Home":
+        event.preventDefault();
+        dateToFocus.get("day") !== 0
+          ? (dateToFocusTemp = dateToFocusTemp.day(1))
+          : (dateToFocusTemp = innerDate.date(day - 1).day(1));
+        focusDate(dateToFocusTemp);
+        break;
+      case "End":
+        event.preventDefault();
+        dateToFocusTemp.get("day") !== 0 && (dateToFocusTemp = dateToFocusTemp.day(7));
+        focusDate(dateToFocusTemp);
+        break;
+      case " ":
+        event.preventDefault();
+        handleDateSelect(day, "date");
+        break;
+    }
+  };
+
+  return (
+    <DatePicker id="date-picker" ref={ref}>
+      <DatePickerToolbar>
+        <DatePickerToolbarButton
+          onClick={() => setContent((content) => (content === "yearPicker" ? "calendar" : "yearPicker"))}
+        >
+          {selectedDate.format("YYYY")}
+        </DatePickerToolbarButton>
+        <DatePickerToolbarSubtitleButton
+          onClick={() => setContent((content) => (content === "monthPicker" ? "calendar" : "monthPicker"))}
+        >
+          {selectedDate.format("MMMM")}
+        </DatePickerToolbarSubtitleButton>
+      </DatePickerToolbar>
+      {content === "calendar" && (
+        <Calendar>
+          <PickerHeader>
+            <PickerHeaderButton
+              aria-label="previous month"
```

Comment:
Icon buttons (buttons without text) need to a have `title` defined for accessibility standards. It should have the same string as the `aria-label`. Also, these inner texts need to be part of our translation strategy, look at the Clear action of the Select for a better understanding.